src/setup/node.rs

```diff
@@ -249,6 +266,14 @@ impl Node {
     pub fn addr(&self) -> SocketAddr {
         self.config.local_addr
     }
+
+    pub fn rpc_url(&self) -> String {
+        format!(
+            "http://{addr}:{port}",
+            addr = self.config.local_addr.ip(),
+            port = JSON_RPC_PORT
+        )
```

Comment:
Well. This is one solution for providing correct RPC URLs for node1 and node2.
But we need to provide the RPC URL everywhere. Not the fanciest solution.

An alternative could be:

- Move the URL RPC address into the RPC object. (rpc.rs)
- Make the RPC object part of the Node struct.
- Then use RPC calls directly on the node via RPC objects in tests like this (example for status.rs):

```
    let mut node = Node::builder()
        .start(target.path(), NodeType::Stateful)
        .await
        .expect("unable to start stateful node");
    node.rpc().wait_for_state("proposing".into()).await;

    let account_data = node.rpc().wait_for_account_data("rNGknFCRBZguXcPqC63k6xTZnonSe6ZuWt")
        .await
        .expect("unable to get account data");
    assert_eq!(account_data.result.account_data.balance, "5000000000");
```

Reply:
@ljedrz What do you think?

Reply:
I don't have strong feelings one way or the other; we can always improve on this later on, if need be.

Reply:
Then let's keep it like this for now.